ellipsoid/v1/caesar/spheroid.hpp

```diff
@@ -0,0 +1,178 @@
+#pragma once
+
+#include <cmath>
+
+namespace caesar {
+
+/**
+ * An ellipsoid with two axes of equal length
+ *
+ * A spheroid is parameterized by its semi-major and semi-minor axis lengths,
+ * \f$a\f$ and \f$b\f$ as
+ *
+ * \f[
+ * \frac{x^2 + y^2}{a^2} + \frac{z^2}{b^2} = 1
+ * \f]
+ *
+ * In geodesy, a spheroid is typically used as a geodetic reference ellipsoid
+ * that approximates the size and shape of the Earth. In such applications, the
+ * semi-major axis length describes the equatorial radius and the semi-minor
+ * axis length describes the polar radius.
+ *
+ * Reference ellipsoids are commonly specified by their semi-major axis and
+ * flattening constant, \f$f\f$, defined as
+ *
+ * \f[
+ * f = \frac{a - b}{a}
+ * \f]
+ *
+ * The flattening is a measure of the ellipticity of the spheroid. If the
+ * flattening is between 0 and 1, the spheroid is oblate (\f$a > b\f$). If it's
+ * equal to 0, the spheroid is a perfect sphere (\f$a == b\f$). If it's less
+ * than zero, the spheroid is prolate (\f$a < b\f$).
+ *
+ * \see wgs84_ellipsoid
+ */
+class Spheroid {
+public:
+    /**
+     * Construct a new Spheroid object.
+     *
+     * \param[in] a semi-major axis length
+     * \param[in] f flattening constant
+     */
+    constexpr Spheroid(double a, double f) noexcept : a_(a), f_(f) {}
+
+    /**
+     * Return the semi-major axis length.
+     *
+     * In geodetic applications, this is the equatorial radius.
+     *
+     * \see Spheroid::a
+     */
+    [[nodiscard]] constexpr double
+    semimajor_axis() const noexcept
```

Comment:
Hmm, this isn't quite right either. By definition, `semimajor_axis` should be the larger of `a` or `b`. Setting it always equal to `a` only works for oblate spheroids. 

Seems like [some celestial objects are actually prolate spheroids](https://en.wikipedia.org/wiki/Spheroid#Prolate_spheroids), so it'd be worthwhile making this work correctly. I'll have to update the documentation, too.